Mathlib/Order/SuccPred/Basic.lean

```diff
@@ -907,13 +983,13 @@ namespace WithTop
 
 section Succ
 
-variable [DecidableEq α] [PartialOrder α] [OrderTop α] [SuccOrder α]
+variable [DecidableEq α] [PartialOrder α] [SuccOrder α]
 
 instance : SuccOrder (WithTop α) where
   succ a :=
     match a with
     | ⊤ => ⊤
-    | Option.some a => ite (a = ⊤) ⊤ (some (succ a))
+    | Option.some a => ite (succ a = a) ⊤ (some (succ a))
```

Comment:
Why doesn't the old definition work anymore?

Reply:
As [mentioned](https://github.com/leanprover-community/mathlib4/pull/15881#issuecomment-2296312137) above, using the new definition the instance can be generalized to avoid the `OrderTop` assumption.